lib/src/model/library.dart

```diff
@@ -281,15 +312,36 @@ class Library extends ModelElement
     return baseName;
   }
 
-  /// Generate a name for this library based on its location.
-  ///
-  /// nameFromPath provides filename collision-proofing for anonymous libraries
-  /// by incorporating more from the location of the anonymous library into
-  /// the name calculation.  Simple cases (such as an anonymous library in
-  /// 'lib') are the same, but this will include slashes and possibly colons
-  /// for anonymous libraries in subdirectories or other packages.
-  late final String nameFromPath =
-      _getNameFromPath(element, package, _restoredUri);
+  @override
+  String get displayName {
+    var fullName = breadcrumbName;
+    if (fullName.endsWith('.dart')) {
+      const dartExtensionLength = '.dart'.length;
+      return fullName.substring(0, fullName.length - dartExtensionLength);
+    }
+    return fullName;
+  }
+
+  @override
+  String get breadcrumbName {
```

Comment:
Just a thought -- Is it weird that the breadcrumb changes from 
`googleapis > admob.v1 library` to 
`googleapis > admob.v1.dart > ...`? And similar with other parts like `googleapis package` to `googleapis`

We already show that it's `admob.v1 library` in the main title on the page. I just find the change in breadcrumbs a little bit odd?

Reply:
I presented this view to @goderbauer and he liked this functionality. I don't feel strongly either way. Most people seemed very against putting ".dart" everywhere but I like it so it's pretty subjective.

I agree there is something to be said for consistency. But Mike really didn't want the `<h1>` to be "animation.dart" or "animation.dart library"; too clinical maybe; not friendly. But in the breadcrumb that name is very meaningful for answering "where do I find this widget / class / function?"

Reply:
Oh yeah, I meant instead of changing the header to be "animation.dart", I was wondering if we could keep the breadcrumbs consistent as we open it and still keep the `.dart` part.

So when you're on the `admob.v1` page, you'd see `googleapis > admob.v1.dart` instead of `googleapis > admob.v1 library` in the breadcrumb, but the `<h1>` can stay the way it is right now. And then the next breadcrumb for App would line up well being `googleapis > admob.v1.dart > App`.

_shrug_ What we have is already fine, but I'm not fond of the breadcrumbs changing as we click into different parts of the library.

Reply:
> So when you're on the admob.v1 page, you'd see googleapis > admob.v1.dart instead of googleapis > admob.v1 library in the breadcrumb, but the <h1> can stay the way it is right now.

Ah I see. *shrug* yeah Mike commented that he liked this way; I don't want to do a whole UX study on it so I'm happy with this line-up. But 100% with the consideration that we can change these things; I don't think it would be jarring to make a few changes over a few months to the sidebar and breadcrumb text, if it doesn't work well for some packages or api.dart.dev or whatever.

Mike, to double check, WDYT about the breadcrumb text changing, when you move from a library page to an element inside that library?

Reply:
For the h1 heading I very much would like to keep it as `animation library` instead of `animation.dart library`. Animation library is what we call this when we talk amongst ourselves or with customers. Nobody says "oh, you can find this in the animation _dot dart_ library". The docs should reflect this real-world use, IMO.

For the breadcrumps I feel less strongly. They feel more technical to me. If we want to change it there to always show the `.dart` for consistency I am not opposed to that. Now that I am thinking about this some more, I would actually slightly lean towards consistency here.

Reply:
SG! I'll make them consistently 'animation.dart' and 'admob/v1.dart' for example and post new screenshots.

Reply:
Screenshots are updated; this is ready for review.

Reply:
Is the screenshot about the animation library still outdated - or is there a bug? I would have expected to see `animation.dart` in the breadcrumb now.

Reply:
Sorry I hadn't updated the Flutter screenshots (takes so long to generate!); updated now.

Reply:
They look good, thanks for updating them Sam!

Reply:
Agree. Screenshots look great!